projects/components/src/datagrid/interfaces/datagrid-column.interface.ts

```diff
@@ -142,6 +151,11 @@ export interface ContextualButtonConfig<R> {
      * Where the buttons should display on the grid.
      */
     position: ContextualButtonPosition;
+    /**
+     * If the title should be the button label, icon, or both
+     * Defaults to ICON if unset.
+     */
+    titleDisplay?: ButtonTitleDisplay;
```

Comment:
I don't think `title` is a descriptive name. I think the enum could be `TextIcon` and `buttonContents` is a more descriptive name for this property.  This could use a bitmask to simplify logic of the `shouldDisplayText` and `shouldDisplayIcon` view methods